src/Data/Unrestricted/Internal/Instances.hs

```diff
@@ -22,59 +23,55 @@
 -- other classes (for cleanness) in this module to avoid this dependence.
 module Data.Unrestricted.Internal.Instances where
 
+import qualified Data.Functor.Linear.Internal.Applicative as Data
+import qualified Data.Functor.Linear.Internal.Functor as Data
+import Data.List.NonEmpty
+import Data.Monoid.Linear
 import Data.Unrestricted.Internal.Consumable
 import Data.Unrestricted.Internal.Dupable
 import Data.Unrestricted.Internal.Movable
 import Data.Unrestricted.Internal.Ur
-import qualified Data.Functor.Linear.Internal.Functor as Data
-import qualified Data.Functor.Linear.Internal.Applicative as Data
+import Data.V.Linear ()
+import Data.V.Linear.Internal.V (V (..), theLength)
+import qualified Data.Vector as Vector
+import GHC.Int
+import GHC.TypeLits
 import GHC.Types hiding (Any)
 import GHC.Word
-import GHC.Int
-import Data.Monoid.Linear
-import Data.List.NonEmpty
-import qualified Prelude
 import Prelude.Linear.Internal
 import qualified Unsafe.Linear as Unsafe
-import Data.V.Linear ()
-import Data.V.Linear.Internal.V (V(..), theLength)
-import qualified Data.Vector as Vector
-import GHC.TypeLits
+import qualified Prelude
+
+newtype AsMovable a = AsMovable a
+
+instance Movable a => Consumable (AsMovable a) where
+  consume (AsMovable x) =
+    move x & \case
+      Ur x' -> ()
+
+instance Movable a => Dupable (AsMovable a) where
+  dupV (AsMovable x) =
+    move x & \case
+      Ur x' -> Data.pure (AsMovable x')
```

Comment:
Just to outline an unimportant alternative: you could have given an instance `instance Movable a => Movable (AsMovable a)`, and used it to save a little space in the `Consumable` and `Dupable` instances.

It doesn't change the generated code, and the code is still very short as it stands; I just wanted to point out that the possibility exists.

Reply:
Oh, I didn't know it was possible to do so, and it looks clearer to me with `instance Movable a => Movable (AsMovable a)`, so I'll add it